src/librustc_mir/transform/type_check.rs

```diff
@@ -59,7 +59,7 @@ pub fn type_check<'a, 'gcx, 'tcx>(
 }
 
 fn mirbug(tcx: TyCtxt, span: Span, msg: &str) {
-    tcx.sess.diagnostic().span_bug(span, msg);
+    tcx.sess.diagnostic().delay_span_bug(span, msg);
```

Comment:
maybe worth a comment here -- something like:

```
// We sometimes see MIR failures (notably predicate failures) due to
// the fact that we check rvalue sized predicates here. So use `delay_span_bug`
// to avoid reporting bugs in those cases.
```